### PR TITLE
Document tag tool, tag anchor block, and groundlevel tag

### PR DESCRIPTION
**Pull request by uecasm**
Depends on ldtteam/minecolonies#7595

## Changes proposed:
- As it says on the tin

Merge please :D

### CODE REVIEW

**Comment by uecasm** [comment]
FYI I deliberately didn't mention some additional tags used only by raid ships, as you can't really make additional styles for those. Server owners _might_ want to replace them but in that case they can look at the existing structures.

File: source/items/tagtool.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+1. Shift-right-click the anchor block of the schematic that you want to view or edit. This will highlight it with a red border and make the associated tags visible.
+
+    * For a building with a hut block, the hut block is the anchor block.
+    * For an upgradeable decoration, the [decoration controller](../../source/items/decocontroller) is the anchor block.
+    * For any other decoration, you must place a Tag Anchor Block as the anchor block.
+
+2. While you are holding the tag tool, you will see all existing tags overlaying their blocks in-world, and they will be highlighted with a white border.
+
+3. Right-click the tag tool in the air to open its GUI. This allows you to type in the tag that you want to apply or remove.  Close the GUI afterwards. The GUI also shows a list of all currently tagged blocks.
```
Comment (ravenbuilder934) [inline source/items/tagtool.md:24]:
```suggestion
3. Right-click the tag tool in the air to open its GUI. This allows you to type in the tag that you want to apply or remove. Close the GUI afterwards. The GUI also shows a list of all currently tagged blocks.

```
one minor nitpick

File: source/items/tagtool.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Current Tags
+
+The following tags are currently implemented (more may be added from time to time):
+
+* `groundlevel`
+
+    This tag is supported by all huts and decorations. It is optional. When included, it should only appear on exactly one block in the schematic that is at the nominal "ground level" of the building -- i.e. any blocks below this should be an underground "basement" area, blocks at the same level should be dirt/grass/other ground blocks (often [solid placeholders](../../source/items/placeholderblocks) are used for this, while underground can be either solid or light placeholders outside of the intended basement area), and blocks above should be the actual building. (If it is specified on more than one block, then only one will take effect. This is harmless if they're all on the same y level but can be confusing if not; hence the recommendation to stick with one at most.)
```
Comment (ravenbuilder934) [inline source/items/tagtool.md:49]: tabs aren't needed at the beginning of paragraphs

File: source/items/tagtool.md (hunk 1 above)
Comment (uecasm) [reply source/items/tagtool.md:24]: Double spacing is the One True Way™.  You younguns that didn't grow up with typewriters (grumble grumble)

File: source/items/tagtool.md (hunk 2 above)
Comment (uecasm) [reply source/items/tagtool.md:49]: It is, when you want the paragraph to be a child of the bullet point and not separate.  At least if this is following normal markdown formatting rules.

**Review by ravenbuilder934 — APPROVED** [review]

**Pull request merged**